src/adapter/src/catalog/migrate.rs

```diff
@@ -90,6 +101,58 @@ pub(crate) async fn migrate<S: Append>(catalog: &mut Catalog<S>) -> Result<(), a
 // AST migrations -- Basic AST -> AST transformations
 // ****************************************************************************
 
+// We previously retained table information for all tables in an upstream PG
+// source's publication, even if the source did not use those tables. In the
+// work of supporting unsupported types as text, it provided a superiror UX to
```

Comment:
I'm curious about the ways in which the UX was inferior if we keep all the information. I was thinking that we might want to keep all the publication info in our catalog so that we can easily support `ALTER SOURCE` statements in the future that add or remove tables

Reply:
With the way the current schema validation worked, if a publication contained an unsupported type, we would error, e.g. when generating the snapshots for all tables in the publication.

Sounds like there's a compelling reason to keep the data around, so will reconsider the approach and leave the entire publication's schema in the `DETAILS`.